### PR TITLE
github: unify PR description and commit message

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

To improve the quality of commit messages in the revision history, indicate that the commit message will be part of the PR review process and integrate the WHAT and WHY questions in the change explanation.

Remove the checklist item for PR title and description as it becomes redundant with this new text.

## Why is it important?

Included in the text above.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.
- [ ] I have made my commit title and message explanatory about the purpose and the reason of the change

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -8,16 +8,25 @@ Please label this PR with one of the following labels, depending on the scope of
 - Docs
 -->
 
-## What does this PR do?
+## Proposed commit message
 
 <!-- Mandatory
-Explain here the changes you made on the PR. Please explain the WHAT: patterns used, algorithms implemented, design architecture, message processing, etc.
--->
+Explain here the changes you made on the PR.
```
Comment (pierrehilbert) [inline .github/PULL_REQUEST_TEMPLATE.md:14]: I would remove this one as we already have "Please explain:" just after.
Otherwise everything is good to me!

Comment (efd6) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: I guess this is a matter of style. The text here reflects the text that was previously there, just now on two lines. The thinking that I had with retaining the text as it is was that the first line is the overall goal of the section, while the second is the specific parts that should be addressed. If you still want me to remove it given that, I'm happy to.

Comment (cmacknz) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: I don't have a problem with it as it is, but if you can make it shorter people are more likely to read it :p

Comment (efd6) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: I agree about that. We need to have reviewers on board to point to this, so the main issue is whether that will happen.

**Review by cmacknz — APPROVED** [review]

**Pull request merged**